api/lib/db-interface.js

```diff
@@ -198,6 +205,16 @@ DI.saveTransaction = function(transaction, callback) {
     txData.rippled_result = result.engine_result;
   }
 
+  var unlockEvent = 'unlocked:' + txData.client_resource_id;
+
+  if (this.lock[txData.client_resource_id]) {
+    // Force synchronous per-transaction updates
+    this.once(unlockEvent, this.saveTransaction.bind(this, transaction, callback));
+    return;
+  } else {
+    this.lock[txData.client_resource_id] = true;
```

Comment:
Non-atomic locking. I guess it works because Node is single-threaded. Relevant discussion: http://stackoverflow.com/questions/2734025/is-javascript-guaranteed-to-be-single-threaded